### ISSUE TITLE
special route cause the page suspend

### ISSUE THREAD

**Issue by snowood-orz** [apache/apisix-dashboard#1922]
## How to Reproduce

I created a 301 route with data
```
{
  "uris": ["/"],
  "name": "root_path",
  "plugins": {
    "redirect": {
      "ret_code": 302,
      "uri": "/login"
    }
  },
  "labels": {
    "tag": "xxx"
  },
  "status": 1
}
```

I clicked the edit button of this route from the route list,the page appeared suspended animation. After a minute, this window Non-stop pop-ups.
```
word in picture:
Please check the service bound to the route. 
If you do not bind the service, you must set up the upstream service (step 2)
```

![image](https://user-images.githubusercontent.com/12845620/119805104-86b19180-bf13-11eb-905c-038eff78edc9.png)

## Environment

all in docker
* apisix version :  apache/apisix:2.5-alpine
* OS: Ubuntu 18.04
* etcd version: bitnami/etcd:3.4.15
* apisix-dashboard version: apache/apisix-dashboard:2.6
* Browser version: 90.0.4430.93 (x86_64)

## Additional context
when choose the redirect,  can not choose the upsteam and plugins.

**Comment by juzhiyuan** [comment]
Just to make sure, the data you mentioned is created by API or dashboard web?

**Comment by starsz** [comment]
> Just to make sure, the data you mentioned is created by API or dashboard web?

I think it's API.

Hi @snowood-orz, maybe you can create a useless upstream.

**Comment by snowood-orz** [comment]
I create route by dashboard web. I have 4 additional upstream, but when I created this route, the dashboard did not provide a page to select the upstream address.

just two pages.
![image](https://user-images.githubusercontent.com/12845620/119966480-ee341380-bfdd-11eb-8ffc-64e4ced84a47.png)

more: I can reproduce this problem in two different host environments.

**Comment by liuxiran** [comment]
> I create route by dashboard web. I have 4 additional upstream, but when I created this route, the dashboard did not provide a page to select the upstream address.
> 
> just two pages.
> ![image](https://user-images.githubusercontent.com/12845620/119966480-ee341380-bfdd-11eb-8ffc-64e4ced84a47.png)
> 
> more: I can reproduce this problem in two different host environments.

Hi @snowood-orz , this issue could be solved by #1926 , you could take a look whether this solution solve your problem, just try the netlify preview URL in the pr, thanks
